cis194/week3/tien/homework3/src/Golf.hs

```diff
@@ -17,3 +20,40 @@ localMaxima (x:y:z:xs)
   | y > x && y > z = y : localMaxima (y : z : xs)
   | otherwise = localMaxima (y : z : xs)
 localMaxima _ = []
+
+-- Exercise 3 --
+formPairs :: [a] -> [(a, Integer)]
+formPairs = flip zip . repeat $ 1
+
+formGroups :: [Integer] -> [Integer]
+formGroups =
+  concat .
+  drop 1 .
+  transpose .
+  map (\(x, y) -> [x, y]) . assocs . accumArray (+) 0 (0, 9) . formPairs
+
+mapToHistogram :: [Integer] -> [String]
+mapToHistogram =
```

Comment:
Not sure how many chars there would be - but FYI, you can use applicative style as an alternative to `if ... then ... else`:

```haskell
import Data.Bool

-- ...

mapToHistogram :: [Integer] -> [String]
mapToHistogram = map intToString
               where
                 intToString = bool <$> const " " <*> const "*" <*> (>0)
```

Reply:
Thanks @sinogermany. Will check and update. :)

Reply:
@sinogermany updated the style. Thanks for the tip. It looks much neater!